### ISSUE TITLE
Add support for passing in context

### ISSUE THREAD

**Issue by boblauer** [pzavolinsky/react-unit#21]
React's `shallowRenderer.render` takes an optional 2nd parameter, `context`.  It would be nice to be able to pass this context in as the 2nd parameter to `createComponent`.

I have tried to get this working, but my functional programming knowledge isn't nearly good enough to understand how this library works.  I was hoping it might be trivial for you to accomplish, although I will keep working on it in the mean time.

**Comment by robertleeplummerjr** [comment]
I believe this gets us 98% of the way there: https://github.com/pzavolinsky/react-unit/pull/22

**Comment by pzavolinsky** [comment]
Hi @boblauer thanks for pointing this out.

As always, @robertleeplummerjr thanks for that great PR! Already went through it and looks good. I will review it in more detail tonight and merge it before Monday.

> I believe this gets us 98% of the way there

Now you got me curious about that extra 2%...

----
Off topic:
> I have tried to get this working, but my functional programming knowledge isn't nearly good enough to understand how this library works. 

This is a good sign that a rewrite is in order to simplify the code and allow more people to collaborate. I've been using typescript lately and is awesome. I will try to *type* this beast. Once type annotations are in place the world will be a better place.

Cheers!

**Comment by robertleeplummerjr** [comment]
or... es6 it?

**Comment by robertleeplummerjr** [comment]
For some reason the context is failing, probably because it is invalid.  This is the error from the unit test:
```
Failures: 
1) with-context when a component is given a context should receive it accordingly
1.1) Invariant Violation: ReactShallowRenderer render(): Invalid component element.
    Invariant Violation: ReactShallowRenderer render(): Invalid component element.
        at invariant (react-unit/node_modules/react/node_modules/fbjs/lib/invariant.js:38:15)
        at ReactShallowRenderer.render (react-unit/node_modules/react/lib/ReactTestUtils.js:368:83)
        at create (react-unit/dist/react-unit.js:264:21)
        at renderElement (react-unit/dist/react-unit.js:273:10)
        at react-unit/dist/react-unit.js:343:10
        at react-unit/node_modules/ramda/dist/ramda.js:290:35
        at react-unit/node_modules/ramda/dist/ramda.js:54:27
        at react-unit/dist/react-unit.js:441:14
        at react-unit/node_modules/ramda/dist/ramda.js:290:35
        at react-unit/node_modules/ramda/dist/ramda.js:54:27
```

**Comment by pzavolinsky** [comment]
Hi @boblauer and @robertleeplummerjr I merged both `withContext` PRs into master and pushed a new version `1.3.4`. 

Thanks a lot for both PRs, in the end I kept @robertleeplummerjr's `withContext` implementation (not a big fan of `R.__`) and @boblauer's `null` checks (Robert's also supported this but is was implicit through an `Object.assign` and the explicit `if` seamed more, well, explicit).

I removed the method access stuff for now so that we could push this new features as soon as possible.

Thanks again guys!

Cheers!
